### ISSUE TITLE
Does the function of  'evaluate()' change the result?

### ISSUE THREAD

**Issue by lceustc** [huggingface/transformers#1556]
## ❓ Questions & Help

when i run RTE task , and logging steps=50: the result  is:
    gloabl_step 50:  0.8953 
    global_step 100:  0.8953 
    gloabl_step 150:  0.8916
    global_step 200:  0.8736
but when logging steps =100:
    global_step 100:  0.8953  
    global_step 200:  0.8880
when global_step is 200,what cause the difference? i didn't  modify any code

**Comment by LysandreJik** [comment]
Did you set a random seed?

**Comment by lceustc** [comment]
for SEEDS in  99

do
CUDA_VISIBLE_DEVICES=2 python run_glue.py \
        --data_dir '/data/ransformers/data/RTE/' \
        --model_type 'roberta' \
        --model_name_or_path '/data/transformers/examples/pretrained_model/roberta-mnli2/' \
        --task_name 'rte' \
        --output_dir ./$SEEDS \
        --overwrite_output_dir \
        --max_seq_length 128 \
        --do_train \
        --do_eval \
        --evaluate_during_training \
        --per_gpu_train_batch_size 8 \
        --per_gpu_eval_batch_size 8 \
        --gradient_accumulation_steps 2 \
        --learning_rate 1e-5 \
        --num_train_epochs 10 \
        --logging_steps 50 \
        --save_steps -1 \
        --seed $SEEDS \

done

**Comment by lceustc** [comment]
> Could you specify what script you're running, with which parameters? Did you set a random seed?




for SEEDS in  99

do
CUDA_VISIBLE_DEVICES=2 python run_glue.py \
        --data_dir '/data/lce/pytorch-transformers/data/RTE/' \
        --model_type 'roberta' \
        --model_name_or_path '/data/lce/pytorch-transformers/examples/pretrained_model/roberta-mnli2/' \
        --task_name 'rte' \
        --output_dir ./$SEEDS \
        --overwrite_output_dir \
        --max_seq_length 128 \
        --do_train \
        --do_eval \
        --evaluate_during_training \
        --per_gpu_train_batch_size 8 \
        --per_gpu_eval_batch_size 8 \
        --gradient_accumulation_steps 2 \
        --learning_rate 1e-5 \
        --num_train_epochs 10 \
        --logging_steps 50 \
        --save_steps -1 \
        --seed $SEEDS \

done

**Comment by lceustc** [comment]
> Could you specify what script you're running, with which parameters? Did you set a random seed?

for SEEDS in  99

do
CUDA_VISIBLE_DEVICES=2 python run_glue.py \
        --data_dir '/data/transformers/data/RTE/' \
        --model_type 'roberta' \
        --model_name_or_path '/data/transformers/examples/pretrained_model/roberta-mnli/' \
        --task_name 'rte' \
        --output_dir ./$SEEDS \
        --overwrite_output_dir \
        --max_seq_length 128 \
        --do_train \
        --do_eval \
        --evaluate_during_training \
        --per_gpu_train_batch_size 8 \
        --per_gpu_eval_batch_size 8 \
        --gradient_accumulation_steps 2 \
        --learning_rate 1e-5 \
        --num_train_epochs 10 \
        --logging_steps 50 \
        --save_steps -1 \
        --seed $SEEDS \

done

**Comment by lceustc** [comment]
> Could you specify what script you're running, with which parameters? Did you set a random seed?

i change 'roberta' to 'bert' and set the same seed, the result is also different,  Is there any wrong with my shell script?
